addons/product/models/product_template.py

```diff
@@ -1476,3 +1476,7 @@ def _get_list_price(self, price):
         To be overridden in accounting module."""
         self.ensure_one()
         return price
+
+    @api.model
+    def _upfront_payment_blacklist(self):
```

Comment:
```suggestion
    def _upfront_payment_blacklist(self):
        """ Service tracking field is used to distinguish some specific categories of products.
        
        Those products shouldn't be displayed or used in unrelated applications.
        This method returns a domain targeting all those specific products (events, courses, ...).
        """
```

Reply:
Also, might make sense to use a more generic method name, I wouldn't be surprised if it was used in other modules/scopes in the future